presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1254,10 +1255,24 @@ private boolean isExpensiveUnionDistinct(SetOperation setOperation, Type[] outpu
                                             type instanceof RowType);
         }
 
+        @Override
+        protected Scope visitUnion(Union node, Optional<Scope> scope)
+        {
+            if (!node.isDistinct().isPresent()) {
+                warningCollector.add(new PrestoWarning(PARSER_WARNING,
+                        "UNION specified without ALL or DISTINCT" +
+                        " keyword is equivalent to UNION DISTINCT, which is computationally expensive. " +
+                        "Consider using UNION ALL when possible, or specifically add keyword DISTINCT to stop " +
```

Comment:
Let's not give the option to add DISTINCT here. So encourage them to add ALL. Most people would just do it. And if it's really needed they will figure it out or ask us. So reword it as something like:

"UNION without the ALL keyword is a computationally expensive operation and most often unnecessarry. Please add the ALL keyword."

Reply:
That's too much. Do you really want the community to open issues here to ask how to make this warning go away?

Reply:
Let's change it to "Consider using UNION ALL when possible, or specifically add the keyword DISTINCT if absolutely necessary".

Reply:
> Let's change it to "Consider using UNION ALL when possible, or specifically add the keyword DISTINCT if absolutely necessary".

Done.